cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1269,15 +1302,18 @@ CacheAllocator<CacheTrait>::findEviction(PoolId pid, ClassId cid) {
       // recycle the candidate.
       if (ReleaseRes::kRecycled ==
           releaseBackToAllocator(itemToRelease, RemoveContext::kEviction,
-                                 /* isNascent */ false, candidate)) {
-        return candidate;
+                                 /* isNascent */ false, toRecycle)) {
+        return toRecycle;
+      }
+    } else if (ref == 0u) {
+      // it's safe to recycle the item here as there are no more
```

Comment:
is this true? If `toReleaseHandle` is empty, we could have returned early from `evictNormalItem()` and didn't remove this item from access-container or mm-container.

Reply:
@igchor: ah I see. Yes you're right. And this is a good point too. In fact, after we have called unmarkMoving, if the ref is 0, then we must free/recycle the memory. Because whichever other thread that has removed it couldn't free the memory due to moving bit.